src/os/ObjectStore.h

```diff
@@ -1716,71 +1716,45 @@ class ObjectStore {
     static void generate_test_instances(list<Transaction*>& o);
   };
 
-  struct C_DeleteTransaction : public Context {
-    ObjectStore::Transaction *t;
-    C_DeleteTransaction(ObjectStore::Transaction *tt) : t(tt) {}
-    void finish(int r) {
-      delete t;
-    }
-  };
-  template<class T>
-  struct C_DeleteTransactionHolder : public Context {
-    ObjectStore::Transaction *t;
-    T obj;
-    C_DeleteTransactionHolder(ObjectStore::Transaction *tt, T &obj) :
-      t(tt), obj(obj) {}
-    void finish(int r) {
-      delete t;
-    }
-  };
-
   // synchronous wrappers
   unsigned apply_transaction(Sequencer *osr, Transaction& t, Context *ondisk=0) {
-    list<Transaction*> tls;
-    tls.push_back(&t);
+    vector<Transaction> tls;
```

Comment:
@cbodley The assumption in this interface was user will not be using the transaction object afterwards...But, agreed that interface should be explicitly saying that it moved..
I will do that , but, I saw it's been used lot of places :-(
Regarding empty after move , that's interesting..
Any idea why it is not able to move TransactionData ?